enhancements/network/user-defined-network-for-virt-workloads.md

```diff
@@ -0,0 +1,563 @@
+	Status IPAMClaimStatus `json:"status,omitempty"`
+}
+
+type IPAMClaimSpec struct {
+	// The network name for which this persistent allocation was created
+	Network string `json:"network"`
+	// The pod interface name for which this allocation was created
+	Interface string `json:"interface"` // *new* attribute
+	// The IP requested by the user
+	// +optional
+	IPRequest []net.IP `json:"ipRequest,omitempty"`
+}
+
+// IPAMClaimStatus contains the observed status of the IPAMClaim.
+type IPAMClaimStatus struct {
+	// The list of IP addresses (v4, v6) that were allocated for the pod interface
+	IPs []string `json:"ips"`
+	// The name of the pod holding the IPAMClaim
+	OwnerPod string `json:"ownerPod"`                            // *new* attribute
+	// Conditions contains details for one aspect of the current state of this API Resource
+	Conditions []metav1.Condition `json:"conditions,omitempty"`  // *new* attribute
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type IPAMClaimList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []IPAMClaim `json:"items"`
+}
+```
+
+This CRD is owned by the k8snetworkplumbingwg; updating it requires quorum from
+the community.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+The management cluster should have no reason to use multiple networks, unless
+it for security reasons it makes sense to use native network isolation over
+network policy.
+
+It makes more sense that multiple primary networks will be used in the hosted
+cluster in order to provide tenants with better isolation from each other,
+without the need for network policy.
+There should be no hypershift platform-specific considerations with this
+feature.
+
+#### Standalone Clusters
+
+Full support.
+
+#### Single-node Deployments or MicroShift
+
+OpenShift Virtualization is not supported on `MicroShift` clusters, hence,
+we do not plan to integrate with UDN on these type of clusters right now.
+
+It is tracked in this [jira issue](https://issues.redhat.com/browse/RFE-5289).
+
+Still, the passt integration is known for imposing some additional overhead
+which might not be suited for MicroShift. This could be mitigated by using
+bridge binding instead, which is more conservative in its resource consumption.
+
+### Implementation Details/Notes/Constraints
+
+N/A
+
+### Risks and Mitigations
+
+Some packets will definitely be dropped during live-migration of the VM; while
+we want to have the migration as seamless as possible, just preserving the
+5 tuple of the existing connections is far from optimal.
+
+If the tests show that the traffic disruption takes too long, we should
+prioritize the
+[seamless live-migration epic](https://issues.redhat.com/browse/CNV-27147)
+which makes use of the multi requested-chassis OVN feature, in which a single
+logical switch port can be bound to multiple chassis (and have traffic
+forwarded to them).
+
```

Comment:
Should we also add here the passt issues ? 
- live migration break connections
- no DHCP hostname

Reply:
the latter is a non-issue to the requirements we have.

the former has been analyzed in the proposal section, and an alternative was provided. I think that's more than enough.